core/src/main/java/com/linecorp/armeria/server/Server.java

```diff
@@ -397,6 +399,18 @@ public String toString() {
                           .toString();
     }
 
+    /**
+     * Reconfigure Server configuration.
+     */
+    public void reconfigure(ServerConfigurator serverConfigurator) {
+        requireNonNull(serverConfigurator, "serverConfigurator");
+        requireNonNull(pipelineConfigurator, "pipelineConfigurator");
+        final ServerBuilder sb = builder();
+        serverConfigurator.reconfigure(sb);
+        config = sb.buildServerConfig(config());
```

Comment:
We need to call `ServiceCallbackInvoker.invokeServiceAdded()` if services are registered to a server. 
https://github.com/line/armeria/blob/4de44afc3017d7153e9bde67f8ff51e855877da1/core/src/main/java/com/linecorp/armeria/server/Server.java#L133-L135
Otherwise, the service will not work as expected.
https://github.com/line/armeria/blob/dbfc1cdbbacb7c15d8128b5b40972de28a82930e/grpc/src/main/java/com/linecorp/armeria/server/grpc/FramedGrpcService.java#L268-L272

Reply:
We also need to think about how we could notify the services being removed. Let me create a separate issue for it.